cirq/work/observable_measurement.py

```diff
@@ -0,0 +1,210 @@
+    "{qubit}-Xf" and "{qubit}-Yf" and are use to change the frame of the
+    qubit before measurement, effectively measuring in bases other than Z.
+    """
+    x_beg_mom = ops.Moment([ops.X(q) ** sympy.Symbol(f'{q}-Xi') for q in qubits])
+    y_beg_mom = ops.Moment([ops.Y(q) ** sympy.Symbol(f'{q}-Yi') for q in qubits])
+    x_end_mom = ops.Moment([ops.X(q) ** sympy.Symbol(f'{q}-Xf') for q in qubits])
+    y_end_mom = ops.Moment([ops.Y(q) ** sympy.Symbol(f'{q}-Yf') for q in qubits])
+    meas_mom = ops.Moment([ops.measure(*qubits, key='z')])
+    if no_initialization:
+        total_circuit = circuit.copy()
+    else:
+        total_circuit = circuits.Circuit([x_beg_mom, y_beg_mom])
+        total_circuit += circuit.copy()
+    total_circuit.append([x_end_mom, y_end_mom, meas_mom])
+    return total_circuit
+
+
+_PAULI_TO_PARAM_VAL: Dict[Tuple['cirq.Pauli', bool], Tuple[float, float]] = {
+    (ops.X, False): (0, -1 / 2),
+    (ops.X, True): (0, +1 / 2),
+    (ops.Y, False): (1 / 2, 0),
+    (ops.Y, True): (-1 / 2, 0),
+    (ops.Z, False): (0, 0),
+    (ops.Z, True): (1, 0),
+}
+
+_STATE_TO_PARAM_VAL: Dict['_NamedOneQubitState', Tuple[float, float]] = {
+    value.KET_PLUS: (0, +1 / 2),
+    value.KET_MINUS: (0, -1 / 2),
+    value.KET_IMAG: (-1 / 2, 0),
+    value.KET_MINUS_IMAG: (+1 / 2, 0),
+    value.KET_ZERO: (0, 0),
+    value.KET_ONE: (1, 0),
+}
+
+
+def _get_params_for_setting(
+    setting: InitObsSetting,
+    flips: Iterable[bool],
+    qubits: Sequence['cirq.Qid'],
+    no_initialization: bool,
+) -> Dict[str, float]:
+    """Return the parameter dictionary for the given setting.
+
+    This must be used in conjunction with a circuit generated by
+    `_with_parameterized_layers`. `flips` (used for readout symmetrization)
+    should be of the same length as `qubits` and will modify the parameters
+    to also include a bit flip (`X`). Code responsible for running the
+    circuit should make sure to flip bits back prior to analysis.
+
+    Like `_with_parameterized_layers`, we omit params for initialization gates
+    if we know that `setting.init_state` is the all-zeros state and
+    `no_initialization` is set.
+    """
+    params = {}
+    for qubit, flip in itertools.zip_longest(qubits, flips):
+        if qubit is None or flip is None:
+            raise ValueError("`qubits` and `flips` must be equal length")
+        # When getting the one-qubit state / observable for this qubit,
+        # you may be wondering what if there's no observable specified
+        # for that qubit. We mandate that by the time you get to this stage,
+        # each _max_setting has
+        # weight(in_state) == weight(out_operator) == len(qubits)
+        # See _pad_setting
+        pauli = setting.observable[qubit]
+        xf_param, yf_param = _PAULI_TO_PARAM_VAL[pauli, flip]
+        params[f'{qubit}-Xf'] = xf_param
+        params[f'{qubit}-Yf'] = yf_param
+
+        if not no_initialization:
```

Comment:
Ohhh double negative. If we keep this - maybe consider renaming to `prepare_state`?

But also, if the init state is well defined anyway, we could just do this always.

Reply:
I think it complains if you try to set parameters that don't exist in the circuit

Reply:
changed the name of this parameter to `needs_init_layer` and inverted the logic